packages/block-editor/src/components/block-lock/toolbar.js

```diff
@@ -33,10 +34,23 @@ export default function BlockLockToolbar( { clientId, wrapperRef } ) {
 	useEffect( () => {
 		if ( isFirstRender.current ) {
 			isFirstRender.current = false;
+			hasModalOpened.current = false;
 			return;
 		}
 
-		if ( ! isModalOpen && shouldHideBlockLockUI ) {
+		if ( isModalOpen && ! hasModalOpened.current ) {
+			hasModalOpened.current = true;
+		}
+
+		// We only want to allow this effeect to happen if the modal has been opened.
+		// The issue is when we're returning focus from the block lock modal to a toolbar,
+		// so it can only happen after a modal has been opened. Without this, the toolbar
+		// will steal focus on rerenders.
```

Comment:
Based on this comment, I'm wondering if this fix will cover all use-cases. What happens if the modal has been opened and closed, could this effect trigger at another moment after the modal was closed (not right after).

Reply:
I agree this isn't a great fix. Definitely a hotfix at best. It's better than the current check, since it at least only surfaces in a rarer instance than the current check.

Rather than try to guess if we need to force focus to the toolbar (and where to focus), I think a better solution would be:

**If the Lock Button has rendered, force it to stay in the toolbar until it is unmounted as a "lock" action button.**

This would mean there would be an "Unlocked" icon until the block toolbar is has been unmounted, but it would improve the UX since the modal could always send focus back to the button that triggered it (and fix this focus stealing issue). Clicking the "Unlocked" icon would trigger the lock modal again.

<img width="502" alt="Block toolbar with an 'unlocked' icon" src="https://github.com/WordPress/gutenberg/assets/967608/512608b6-d272-471d-89f5-9e557e09dec2">

Reply:
Agreed — this sounds like a good idea that we should try out separately from this hot fix.